src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -80,28 +80,30 @@ def queryDatabase(self, query, projection=None):
         projection = {}
         for key in query.get('filters', []):
             projection[key] = 1
-        return self.mgr.getPileup(spec, projection)
+        return self.dataMgr.getPileup(spec, projection)
 
     def createPileup(self, pdict):
         """
         MSPileup create pileup API to create appropriate pileup document
         in underlying database.
 
-        :param pdict: input MSPilup data dictionary
+        :param pdict: input MSPileup data dictionary
         :return: results of MSPileup data layer (list of dicts)
         """
         self.authMgr.authorizeApiAccess('ms-pileup', 'create')
-        return self.mgr.createPileup(pdict)
+        rseNames = self.rucio.evaluateRSEExpression(self.diskRSEExpr, useCache=True)
+        return self.dataMgr.createPileup(pdict, rseNames)
 
     def updatePileup(self, pdict):
         """
-        MSPileup update API to update correspoding pileup document in data layer.
+        MSPileup update API to update corresponding pileup document in data layer.
 
         :param pdict: input MSPilup data dictionary
         :return: results of MSPileup data layer (list of dicts)
         """
         self.authMgr.authorizeApiAccess('ms-pileup', 'update')
-        return self.mgr.updatePileup(pdict)
+        rseNames = self.rucio.evaluateRSEExpression(self.diskRSEExpr, useCache=True)
```

Comment:
Alan, I thought we agree to not use the cache, while here it is the opposite. Could you please elaborate on this `useCache=True` option.

Reply:
No, we actually agreed to use the Rucio wrapper cache, specially because this call is tied to EVERY HTTP call, so we better use it not to increase latency in the HTTP calls (also because the RSE expression is always the same!).

Reply:
The discussion was about effect of the cache to HTTP service which should be simple. Said that, I will not argue here until we know that cache really significantly contribute to RAM usage. Let's proceed with `usageCache=True` and keep eye on RAM usage.

Reply:
I don't think we need to be concerned with this because cache will be a dictionary with a single key/value pair. Any updates to the cache will be void because the key will already exist.